src/Video/video.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useRef, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faMicrophoneSlash,
+  faMicrophone,
+  faVideo,
+  faVideoSlash,
+} from '@fortawesome/free-solid-svg-icons';
+
+const VideoContainer = styled.div`
+  position: absolute;
+  z-index: 999;
+  display: flex;
+  flex-direction: column;
+`;
+
+const Video = styled.video`
+  height: ${(props) => props.height || 150}px;
+  width: ${(props) => props.width || 150}px;
+  background-color: black;
+`;
+
+const IconContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  background-color: rgb(220, 220, 220, 0.5);
+  border-radius: 32px;
+`;
+
+const MicrophoneSlash = styled(FontAwesomeIcon)`
+  color: #ed4650;
+  margin: 2%;
+`;
+
+const Microphone = styled(FontAwesomeIcon)`
+  color: black;
+  margin: 2%;
+`;
+
+const VideoIcon = styled(FontAwesomeIcon)`
+  color: black;
+  margin: 2%;
+`;
+
+const VideoIconSlash = styled(FontAwesomeIcon)`
+  color: #ed4650;
+  margin: 2%;
+`;
+
+type VideoProps = {
+  height?: number;
+  width?: number;
+};
+
+enum VideoAudioOptions {
+  audio = 'audio',
+  video = 'video',
+  all = 'all',
+}
```

Comment:
```suggestion
enum AVOptions {
  AUDIO,
  VIDEO,
  BOTH,
}
```

Also you can just index into the enum to get a string value, no need to assign it (`AVOptions[AVOptions.AUDIO]`). Also for enums you can use a switch statement or compare it by the actual variant `trackKind === AVOptions.BOTH`. Don't compare the string value itself. That defeats the whole point of using enums.

Reply:
Thanks for the feedback here!  Still new to using `enums` and this is great. 

I'd note that at least in Typescript, `string enums` are supported.  They're useful when debugging, or when you'd like  a meaningful and readable value when your code runs, independent of the name of the enum member itself.
https://www.typescriptlang.org/docs/handbook/enums.html#string-enums

Ex)
```typescript
enum Direction {
  Up = "UP",
  Down = "DOWN",
  Left = "LEFT",
  Right = "RIGHT",
}
```

That said, I think it makes sense to refactor based on the suggestion you gave!  So you should see that reflected in the PR.